common/beerocks/bcl/source/network/network_utils.cpp

```diff
@@ -363,6 +365,95 @@ static int parseRoutes(struct nlmsghdr *nlHdr, std::shared_ptr<route_info> rtInf
     return (0);
 }
 
+std::shared_ptr<std::unordered_map<std::string, std::string>>
+network_utils::get_arp_table(bool mac_as_key)
+{
+    auto arp_table_ptr = std::make_shared<std::unordered_map<std::string, std::string>>();
+    auto &arp_table    = *arp_table_ptr.get();
+
+    // Allocate a new netlink socket
+    auto nl_socket = nl_socket_alloc();
+    if (nl_socket == nullptr) {
+        LOG(ERROR) << "Failed allocating netlink socket!";
+        return nullptr;
+    }
+
+    int err = nl_connect(nl_socket, NETLINK_ROUTE);
+    if (err != 0) {
+        LOG(ERROR) << "Failed connecting the netlink socket: " << nl_geterror(err);
+        nl_socket_free(nl_socket);
+        return nullptr;
+    }
+
+    // ARP table cache
+    struct nl_cache *neighbor_table_cache = nullptr;
+
+    // Allocate the cache and fill it with data
+    err = rtnl_neigh_alloc_cache(nl_socket, &neighbor_table_cache);
+    if (err != 0) {
+        LOG(ERROR) << "Failed probing ARP table: " << nl_geterror(err);
+        nl_socket_free(nl_socket);
+        return nullptr;
+    }
+
+    nl_socket_free(nl_socket);
+
+    auto enteries_count = nl_cache_nitems(neighbor_table_cache);
+
+    constexpr const char *error_message = "neighbor_entry is null";
```

Comment:
What? Why?

Reply:
I did it because I didn't want to create the same string in two locations.
If what you wonder about is the `constexpr const` - I wanted that the string itself be a const.
This is equivalent to `char const * const`.